internal/database/mariadb/database.go

```diff
@@ -165,8 +165,14 @@ func buildFilterQuery[T any](filter []T, expr string, op string) string {
 }
 
 func buildQueryParameters[T any](params []interface{}, filter []T) []interface{} {
+	return buildQueryParametersCount(params, filter, 1)
+}
+
+func buildQueryParametersCount[T any](params []interface{}, filter []T, count int) []interface{} {
 	for _, item := range filter {
-		params = append(params, item)
+		for i := 0; i < count; i++ {
+			params = append(params, item)
+		}
```

Comment:
Can you explain why we need to do this?

Reply:
This is used because Wild Card query has multiple logical statements with the same parameter, here is pseudo code we are using:
`'WHERE ....some previous conditions... AND (primName LIKE searchParam OR secName LIKE searchParam) AND ...`
`
search param is the same for every single element from search list of filters. It is passed currently through two SQL Query parameters passed by question marks per search element. In all other filters, we have only single question marks used.
We could add extra SET Sql keyword to set every parameter to some value but it will complicate the design a bit, but I am open for discussion.

Meanwhile I will try to find better name for this function I have added to database.go.

Reply:
Ahh thanks for explaining